storybookjs/storybook
# Update glob to v10.0.0

## What I did

Updated the `glob` library to v10, which includes a promise-based API and a min version of Node 16. Also removed the no longer needed glob-promise library.

## How to test

This updates the logic for listing stories in a vite project. To test:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Check that all expected stories are displayed.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
